code/modules/hippie/pool/pool_controller.dm

```diff
@@ -0,0 +1,363 @@
+					if(drownee.stat != CONSCIOUS)
+						drownee.adjustOxyLoss(9)
+						drownee << "<span class='danger'>You're quickly drowning!</span>"
+					else
+						if(!drownee.internal)
+							drownee.adjustOxyLoss(4)
+							if(prob(35))
+								drownee << "<span class='danger'>You're lacking air!</span>"
+
+			for(var/obj/effect/decal/cleanable/decal in W)
+				if(bloody < 800)
+					animate(decal, alpha = 10, time = 20)
+					spawn(25)
+						qdel(decal)
+				if(istype(decal,/obj/effect/decal/cleanable/blood) || istype(decal, /obj/effect/decal/cleanable/trail_holder))
+					bloody++
+					if(bloody > lastbloody)
+						changecolor()
+
+/obj/machinery/poolcontroller/proc/changecolor()
+	lastbloody = bloody+99
+	switch(bloody)
+		if(0 to 99)
+			for(var/turf/open/pool/water/color1 in linkedturfs)
+				color1.color = "#FFFFFF"
+				color1.watereffect.color = "#FFFFFF"
+		if(100 to 199)
+			for(var/turf/open/pool/water/color1 in linkedturfs)
+				color1.color = "#FFDDDD"
+				color1.watereffect.color = "#FFDDDD"
+		if(100 to 199)
+			for(var/turf/open/pool/water/color1 in linkedturfs)
+				color1.color = "#FFCCCC"
+				color1.watereffect.color = "#FFCCCC"
+		if(200 to 299)
+			for(var/turf/open/pool/water/color1 in linkedturfs)
+				color1.color = "#FFBBBB"
+				color1.watereffect.color = "#FFBBBB"
+		if(300 to 399)
+			for(var/turf/open/pool/water/color1 in linkedturfs)
+				color1.color = "#FFAAAA"
+				color1.watereffect.color = "#FFAAAA"
+		if(400 to 499)
+			for(var/turf/open/pool/water/color1 in linkedturfs)
+				color1.color = "#FF9999"
+				color1.watereffect.color = "#FF9999"
+		if(500 to 599)
+			for(var/turf/open/pool/water/color1 in linkedturfs)
+				color1.color = "#FF8888"
+				color1.watereffect.color = "#FF8888"
+		if(600 to 699)
+			for(var/turf/open/pool/water/color1 in linkedturfs)
+				color1.color = "#FF7777"
+				color1.watereffect.color = "#FF7777"
+		if(700 to 799)
+			for(var/turf/open/pool/water/color1 in linkedturfs)
+				color1.color = "#FF7777"
+				color1.watereffect.color = "#FF7777"
+		if(800 to 899)
+			for(var/turf/open/pool/water/color1 in linkedturfs)
+				color1.color = "#FF6666"
+				color1.watereffect.color = "#FF6666"
+		if(900 to INFINITY)
+			for(var/turf/open/pool/water/color1 in linkedturfs)
+				color1.color = "#FF5555"
+				color1.watereffect.color = "#FF5555"
+				src.bloody = 1000
+
+/obj/machinery/poolcontroller/proc/miston() //Spawn /obj/effect/mist (from the shower) on all linked pool tiles
+	for(var/turf/open/pool/water/W in linkedturfs)
+		var/M = new /obj/effect/mist(W)
+		if(misted)
+			return
+		linkedmist += M
+
+	misted = 1 //var just to keep track of when the mist on proc has been called.
+
+/obj/machinery/poolcontroller/proc/mistoff() //Delete all /obj/effect/mist from all linked pool tiles.
+	for(var/obj/effect/mist/M in linkedmist)
+		qdel(M)
+	misted = 0 //no mist left, turn off the tracking var
+
+/obj/machinery/poolcontroller/attack_hand(mob/user)
```

Comment:
Would you mind porting this to the fancy tg-ui? this feels rather clunky, I can help if you want